server/controllers/nlp.js

```diff
@@ -47,12 +208,76 @@ exports.handleInput = function (req, res) {
 
         let dataset = req.body.dataset;
 
+        let lowerCaseInput = req.body.input.toLowerCase();
+
+        var tokenizer = new natural.WordTokenizer();
+        let tokenizedInput = tokenizer.tokenize(lowerCaseInput);
+
+
+        let initState = {
+            tokens: tokenizedInput,
+            layer: 0,
```

Comment:
What is a layer?

Reply:
Layers are the different types of Expressions we search for and then label the words. So for example charttype Column, Connector or operations